engine/src/main/java/org/terasology/math/Edge.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.math;
+
+import org.terasology.naming.Name;
+
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.EnumSet;
+import java.util.Set;
+
+/**
+ * A cube has 12 edges. For asymmetric cubes each edge can have 2 orientations, giving 24 different combinations.
+ * First 12 edges are for symmetric blocks, all 24 are for asymmetric blocks.
+ */
+public enum Edge {
+
+    BottomBack(Rotation.none()),
```

Comment:
The names should follow the java naming convention that constants are upper case and separted with _. Otherwise the names are fine as they match the names of the Side enum.